pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -328,6 +334,8 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 	}
 
 	s.availableOnNodes = make([]*nodeaffinity.NodeSelector, len(claims))
+	s.suitableNodes = make([]*v1.NodeSelector, len(claims))
+	s.unsuitableNodes = make([]sets.String, len(claims))
```

Comment:
We are creating three slices here which all have the same length. I think it'll be better to create one slice holding a struct with the three values (less allocations and fragmentation).

Reply:
After review the code, I think we only need two slices here now, is it still better to create a struct?

Reply:
I think so, because it reduces the amount of allocations with no downsides as far as I can tell.